go/workflow.go

```diff
@@ -26,17 +26,17 @@ func BreakfastWorkflow(ctx workflow.Context, parallelCompensations bool) (err er
 		}
 	}()
 
+	compensations.AddCompensation(PutBowlAway)
 	err = workflow.ExecuteActivity(ctx, GetBowl).Get(ctx, nil)
 	if err != nil {
 		return err
 	}
-	compensations.AddCompensation(PutBowlAway)
 
+	compensations.AddCompensation(PutCerealBackInBox)
 	err = workflow.ExecuteActivity(ctx, AddCereal).Get(ctx, nil)
 	if err != nil {
 		return err
 	}
-	compensations.AddCompensation(PutCerealBackInBox)
 
 	err = workflow.ExecuteActivity(ctx, AddMilk).Get(ctx, nil)
```

Comment:
Admittedly now it feels extra weird to not have a compensation for the AddMilk case, but perhaps this is a good opportunity to remind the reader (of my article) that not every step needs to have a compensation.

Reply:
Agree, especially with this retry policy and if you consider supporting cancellation.
But let's keep as is for now, this is good enough.